app/authentication/authentication.js

```diff
@@ -1,138 +1,39 @@
-        .then((data) => {
-          user = data[0];
-        })
-        .catch((error) => {
-          console.log(error);
-        });
-      if (user != null) {
-        let hash = await hashPassword(password, user.salt);
-        if (Buffer.compare(user.password, hash) !== 0) {
-          return res.status(401).send({
-            message: "Invalid password!",
-          });
-        }
-        return {
-          type: "credentials",
-          userId: user.id,
-        };
-      } else {
-        return res.status(401).send({
-          message: "User not found!",
-        });
-      }
-    }
-    if (
-      auth.startsWith("Bearer ") &&
-      (typeof require !== "string" || require === "token")
-    ) {
-      let token = auth.slice(7);
-      let sessionId = await decrypt(token);
-      let session = {};
-      await Session.findAll({ where: { id: sessionId } })
-        .then((data) => {
-          session = data[0];
-        })
-        .catch((error) => {
-          console.log(error);
-        });
-      if (session != null) {
-        if (session.expirationDate >= Date.now()) {
-          return {
-            type: "token",
-            userId: session.userId,
-            sessionId: session.id,
-          };
-        } else {
-          return res.status(401).send({
-            message: "Session has expired.",
-          });
-        }
-      } else {
-        return res.status(401).send({
-          message: "Invalid session",
-        });
-      }
-    }
-  }
-  if (require) {
-    return res.status(401).send({
-      message: "Authentication required",
+  let email = req.body.email;
+  let password = req.body.password;
+  let user = {};
+  await User.findAll({ where: { email: email } })
+    .then((data) => {
+      user = data[0];
+    })
+    .catch((error) => {
+      console.log(error);
     });
-  }
-  return { type: "none", userId: null };
-};
-
-authenticateRoute = async (req, res, next) => {
-  let auth = req.get("authorization");
-  console.log(auth);
-  if (auth != null) {
-    if (
-      auth.startsWith("Bearer ") &&
-      (typeof require !== "string" || require === "token")
-    ) {
-      let token = auth.slice(7);
-      let sessionId = await decrypt(token);
-      let session = {};
-      await Session.findAll({ where: { id: sessionId } })
-        .then((data) => {
-          session = data[0];
-        })
-        .catch((error) => {
-          console.log(error);
-        });
-      if (session != null) {
-        console.log(session >= Date.now());
-        console.log(Date.now());
-        if (session.expirationDate >= Date.now()) {
-          next();
-          return;
-        } else {
-          return res.status(401).send({
-            message: "Unauthorized! Expired Token, Logout and Login again",
-          });
-        }
-      } else {
-        return res.status(401).send({
-          message: "Unauthorized! Expired Token, Logout and Login again",
-        });
-      }
+  if (user != null) {
```

Comment:
if `user` is still `{}` because there was an error looking up the user by e-mail, this will still be `true` because `{} != null` is `true` in JavaScript